### PR TITLE
test/integration/client: test exec calls metric

### PR DESCRIPTION
**Pull request by ankeesler**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

* This PR adds more integration tests for the client-go credential provider feature.
* We are taking the feature GA, and thus need to implement the test plan from the KEP.
* KEP: https://github.com/kubernetes/enhancements/blob/master/keps/sig-auth/541-external-credential-providers/README.md.

#### Which issue(s) this PR fixes:

* Feature set tracking issue: kubernetes/enhancements#541
* KEP: https://github.com/kubernetes/enhancements/blob/master/keps/sig-auth/541-external-credential-providers/README.md

#### Special notes for your reviewer:

* I didn't add tests for the cert metrics, I can add those next if we want them.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/sig auth

### CODE REVIEW

File: staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go [hunk 1]
```diff
@@ -263,8 +263,9 @@ func (a *Authenticator) UpdateTransportConfig(c *transport.Config) error {
 	// setting up the transport, as that triggers the exec action if the server is
 	// also configured to allow client certificates for authentication. For requests
 	// like "kubectl get --token (token) pods" we should assume the intention is to
-	// use the provided token for authentication.
-	if c.HasTokenAuth() {
+	// use the provided token for authentication. The same can be said for when the
+	// user specifies basic auth.
+	if c.HasTokenAuth() || c.HasBasicAuth() {
```
Comment (enj) [inline staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go:268]: Split this change into its own PR with matching tests like #91745.

File: test/integration/client/testdata/exec-plugin.sh [hunk 2]
```diff
@@ -23,4 +23,6 @@ if [[ -n "${EXEC_PLUGIN_OUTPUT_FILE-""}" ]]; then
   exit
 fi
 
-echo "$EXEC_PLUGIN_OUTPUT"
+echo "${EXEC_PLUGIN_OUTPUT-""}"
+
+exit "${EXEC_PLUGIN_EXEC_CODE-0}"
```
Comment (enj) [inline test/integration/client/testdata/exec-plugin.sh:28]: nit: fix missing newline

File: test/integration/client/exec_test.go [hunk 3]
```diff
@@ -388,6 +472,17 @@ func TestExecPluginViaClient(t *testing.T) {
 	}
 }
 
+func captureMetrics(t *testing.T) *execPluginMetrics {
+	previousCallsMetric := metrics.ExecPluginCalls
+	t.Cleanup(func() {
+		metrics.ExecPluginCalls = previousCallsMetric
+	})
```
Comment (enj) [inline test/integration/client/exec_test.go:479]: 🤮 at the global state.

**Review by enj — CHANGES_REQUESTED** [review]
Can we test the `client_internal_error` case?

File: staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go (hunk 1 above)
Comment (ankeesler) [reply staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go]: Included in 31c53024ff9c54597a7c3da511eb5e864f5e6140. PR: https://github.com/kubernetes/kubernetes/pull/102175/files.

File: test/integration/client/testdata/exec-plugin.sh (hunk 2 above)
Comment (ankeesler) [reply test/integration/client/testdata/exec-plugin.sh]: Included in 31c53024ff9c54597a7c3da511eb5e864f5e6140.

File: test/integration/client/exec_test.go (hunk 3 above)
Comment (ankeesler) [reply test/integration/client/exec_test.go:492]: Same.

**Comment by enj** [comment]
/lgtm cancel
(save CI while tests are fixed)
